lib/i18n_data.rb

```diff
@@ -47,12 +47,13 @@ def data_provider=(provider)
 
     private
 
-    def fetch(*args)
+    def fetch(type, language_code)
       @cache ||= {}
-      if @cache.key?(args)
-        @cache[args]
+      @cache[type] ||= {}
+      if @cache[type].key?(language_code)
+        @cache[type][language_code]
       else
-        @cache[args] = yield
+        @cache[type][language_code] = yield
       end
```

Comment:
how about: 
 - initialize `@cache` as `Hash.new { |h, k| h[k] = {} }`
 - `@cache[type].fetch(language_code) { @cache[language_code] = yield }`

Reply:
That is more concise, and performant, so: yes!
`  0.180000   0.000000   0.180000 (  0.181957)`

Reply:
(Added `[type]` to the contents of the `fetch` block, BTW.)